### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by szymondziedziel**
Some basic setup

### CODE REVIEW

File: client/app/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "extends": [
+    "plugin:@typescript-eslint/recommended",
+    "prettier/@typescript-eslint",
+    "plugin:prettier/recommended"
+  ],
+  "parserOptions": {
+    "ecmaVersion": 2020,
```
Comment (szymondziedziel) [inline client/app/.eslintrc.json:9]: We can consider to use to use lower version here.

File: client/app/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "sourceType": "module"
+  },
+  "ignorePatterns": [
+    "package.json",
+    "package-lock.json",
+    "dist",
+    "node_modules"
+  ],
+  "rules": {}
```
Comment (szymondziedziel) [inline client/app/.eslintrc.json:18]: I left it empty to have all default.

File: client/app/.gitignore [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+# /coverage
+# /libpeerconnection.log
+# npm-debug.log
+# yarn-error.log
+# testem.log
+# /typings
+#
+# # System Files
+# .DS_Store
+# Thumbs.db
```
Comment (szymondziedziel) [inline client/app/.gitignore:246]: This is auto generated by NG CLI and it was not working as expected. I used gitignore.io and it seems to work muhc better

File: client/app/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# App
```
Comment (szymondziedziel) [inline client/app/README.md:1]: This is autogenerated I do not need this personally.

File: client/app/__tslint.json [hunk 5]
```diff
@@ -0,0 +1,104 @@
+{
```
Comment (szymondziedziel) [inline client/app/__tslint.json:1]: This is renamed autogenerated config. I did not want to remove it, that's why I renamed it. Out original is empty object `{}`

File: client/app/angular.json [hunk 6]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (szymondziedziel) [inline client/app/angular.json:1]: This is something which may be improved further by this https://angular.io/guide/workspace-config.

For example I removed test as we moved to JEST, maybe lint should be also removes after adding HTML and SCSS linters

File: client/app/jest.config.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const { pathsToModuleNameMapper } = require('ts-jest/utils');
+const { compilerOptions } = require('./tsconfig');
+
+module.exports = {
+  preset: 'jest-preset-angular',
+  roots: ['./src/'],
+  testMatch: ['**/+(*.)+(spec).+(ts)'],
+  setupFilesAfterEnv: ['./src/test.ts'],
+  collectCoverage: true,
+  coverageReporters: ["text"],
```
Comment (szymondziedziel) [inline client/app/jest.config.js:10]: I put here text only as I do not think we need have to generate and store test reports.

File: client/app/lint-html.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+var validate = require('html-angular-validate');
```
Comment (szymondziedziel) [inline client/app/lint-html.js:1]: This need to be improved by all rules if we want to properly lint HTML. Or maybe we want to get back to built in angular lint.

File: client/app/src/app/components/remove-this-uncovered-component/remove-this-uncovered-component.component.html [hunk 9]
```diff
@@ -0,0 +1,2 @@
+<p>remove-this-uncovered-component works!</p>
```
Comment (szymondziedziel) [inline client/app/src/app/components/remove-this-uncovered-component/remove-this-uncovered-component.component.html:1]: Remove file

File: client/app/src/app/components/remove-this-uncovered-component/remove-this-uncovered-component.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (szymondziedziel) [inline client/app/src/app/components/remove-this-uncovered-component/remove-this-uncovered-component.component.spec.ts:1]: Remove file

File: client/app/src/app/components/remove-this-uncovered-component/remove-this-uncovered-component.component.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { Component } from '@angular/core';
```
Comment (szymondziedziel) [inline client/app/src/app/components/remove-this-uncovered-component/remove-this-uncovered-component.component.ts:1]: Remove file

File: client/app/README.md (hunk 4 above)
Comment (MateuszMyrcik) [reply client/app/README.md:1]: I think we can stay with that

File: client/app/jest.config.js (hunk 7 above)
Comment (MateuszMyrcik) [reply client/app/jest.config.js:10]: Think to discuss on check-up meeting

File: client/app/lint-html.js (hunk 8 above)
Comment (MateuszMyrcik) [reply client/app/lint-html.js:1]: TBD

File: client/app/README.md (hunk 4 above)
Comment (MateuszMyrcik) [reply client/app/README.md:1]: Move to root readme

File: client/app/package.json [hunk 12]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "food-track-radar",
+  "version": "0.1.0",
+  "scripts": {
+    "ng": "ng",
+    "lint": "eslint . --ext ts,json,html --fix --color",
```
Comment (MateuszMyrcik) [inline client/app/package.json:6]: are we sure that lint support checking html?

**Pull request merged**